webapp/components/post_view/components/post_list.jsx

```diff
@@ -50,12 +54,96 @@ export default class PostList extends React.Component {
             this.introText = this.getArchivesIntroMessage();
         }
 
+        let lastViewed = Number.MAX_VALUE;
+        const member = ChannelStore.getMember(props.channel.id);
+        if (member != null) {
+            lastViewed = member.last_viewed_at;
+        }
+
         this.state = {
             isScrolling: false,
-            topPostId: null
+            topPostId: null,
+            lastViewed,
+            ownNewMessage: false
         };
     }
 
+    componentWillReceiveProps(nextProps) {
+        if (this.props.channel !== nextProps.channel) {
+            let lastViewed = Number.MAX_VALUE;
+            const member = ChannelStore.getMember(nextProps.channel.id);
+            if (member != null) {
+                lastViewed = member.last_viewed_at;
+            }
+            this.setState({
+                lastViewed,
+                ownNewMessage: false
+            });
+        }
+    }
+
+    setUnreadPost(post) {
```

Comment:
Can you move this function up into the parent `post_view_controller.jsx`? We're doing out best to only access the stores inside view controllers.

Also can you move any setting of data in the stores and any common functionality into an action through a function in `actions/post_actions.jsx`? That way it can be reused by other components in the future